supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/Run.scala

```diff
@@ -117,22 +117,31 @@ object Run {
   }
 
   def interpretOperationStatus(op: Operation): RunStatus = {
-    if (op.getDone) {
-      lazy val eventList = getEventList(op)
-      lazy val ceInfo = op.getMetadata.get ("runtimeMetadata").asInstanceOf[GArrayMap[String,Object]].get("computeEngine").asInstanceOf[GArrayMap[String, String]]
-      lazy val machineType = Option(ceInfo.get("machineType"))
-      lazy val instanceName = Option(ceInfo.get("instanceName"))
-      lazy val zone = Option(ceInfo.get("zone"))
-
-      // If there's an error, generate a Failed status. Otherwise, we were successful!
-      Option(op.getError) match {
-        case None => Success(eventList, machineType, zone, instanceName)
-        case Some(error) => Failed(error.getCode, Option(error.getMessage), eventList, machineType, zone, instanceName)
+    try {
+      if (op.getDone) {
+        lazy val eventList = getEventList(op)
+        lazy val computeEngineOption = for {
+          runtimeMetadata <- op.getMetadata.asScala.get("runtimeMetadata")
+          computeEngine <- runtimeMetadata.asInstanceOf[GArrayMap[String, Object]].asScala.get("computeEngine")
+        } yield computeEngine.asInstanceOf[GArrayMap[String, String]].asScala
+        lazy val machineType = computeEngineOption.flatMap(_.get("machineType"))
+        lazy val instanceName = computeEngineOption.flatMap(_.get("instanceName"))
+        lazy val zone = computeEngineOption.flatMap(_.get("zone"))
+
+        // If there's an error, generate a Failed status. Otherwise, we were successful!
+        Option(op.getError) match {
+          case None => Success(eventList, machineType, zone, instanceName)
+          case Some(error) =>
+            Failed(error.getCode, Option(error.getMessage), eventList, machineType, zone, instanceName)
+        }
+      } else if (op.hasStarted) {
+        Running
+      } else {
+        Initializing
       }
-    } else if (op.hasStarted) {
-      Running
-    } else {
-      Initializing
+    } catch {
+      case npe: NullPointerException =>
+        throw new RuntimeException(s"Caught NPE while processing operation ${op.getName}: $op", npe)
```

Comment:
`op.getDone` can throw a NPE, as it uses a scala implict conversion to convert from `java.lang.Boolean` to a `boolean`. `getEventList` could also throw an error when trying to call `getMetadata`. There are test specs for both.

Also you're correct that `op` itself could be also be `null`. I'll add a check and test for that too.